autoapi/domains/javascript.py

```diff
@@ -0,0 +1,201 @@
+import os
+import json
+import subprocess
+
+from sphinx.util.osutil import ensuredir
+
+from .base import AutoAPIBase, AutoAPIDomain
+from ..settings import env
+
+
+class JavaScriptDomain(AutoAPIDomain):
+
+    '''Auto API domain handler for Javascript
+
+    Parses directly from Javascript files.
+
+    :param app: Sphinx application passed in as part of the extension
+    '''
+
+    def create_class(self, data):
+        '''Return instance of class based on Javascript data
+
+        Data keys handled here:
+
+            type
+                Set the object class
+
+            consts, types, vars, funcs
+                Recurse into :py:meth:`create_class` to create child object
+                instances
+
+        :param data: dictionary data from godocjson output
+        '''
+        obj_map = dict(
+            (cls.type, cls) for cls
+            in ALL_CLASSES
+        )
+        try:
+            cls = obj_map[data['kind']]
+        except KeyError:
+            self.app.warn('Unknown Type: %s' % data)
+        else:
+            # Recurse for children
+            obj = cls(data)
+            if 'children' in data:
+                for child_data in data['children']:
+                    for child_obj in self.create_class(child_data):
+                        obj.children.append(child_obj)
+            yield obj
+
+    def read_file(self, path, **kwargs):
+        '''Read file input into memory, returning deserialized objects
+
+        :param path: Path of file to read
+        '''
+        # TODO support JSON here
+        # TODO sphinx way of reporting errors in logs?
+
+        try:
+            parsed_data = json.loads(subprocess.check_output(['jsdoc', '-X', path]))
+            return parsed_data
+        except IOError:
```

Comment:
Catch `OSError` or something from `subprocess` here, if `jsdoc` is not installed or in the PATH